### PR TITLE
Add makefile targets to deploy e2e resources without running tests

### PR DESCRIPTION
**Pull request by alexeykazakov**
It will be useful when developing operators and/or registration service:

- `make dev-deploy-e2e-local` - deploys the same resources as `make test-e2e-local` but doesn't run tests.
- `make dev-deploy-e2e` - deploys the same resources as `make test-e2e` but doesn't run tests.

Be default instead of using generated NS names it uses `toolchain-host/member-operator` namespaces.

From reg-service/host/member repos these targets can be re-used to deploy everything using just one command.
For example from reg-service repo: `make deploy-e2e` will download everything but the reg-service itself and deploy it to OpenShift cluster - https://github.com/codeready-toolchain/registration-service/pull/70

### CODE REVIEW

File: make/dev.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Deploy the e2e environment with the local 'host', 'member', and 'registration-service' repositories
+dev-deploy-e2e-local:
+	$(MAKE) deploy-e2e-local MEMBER_NS=${DEV_MEMBER_NS} HOST_NS=${DEV_HOST_NS} REGISTRATION_SERVICE_NS=${DEV_REGISTRATION_SERVICE_NS}
+	@echo "Deployment complete!"
+	@echo "To clean the cluster run 'make clean-e2e-resources'"
+
+.PHONY: dev-deploy-e2e-member-local
+## Deploy the e2e resources with the local 'member-operator' repository only
+dev-deploy-e2e-member-local:
+	$(MAKE) dev-deploy-e2e MEMBER_REPO_PATH=${PWD}/../member-operator
```
Comment (xcoulon) [inline make/dev.mk]: shouldn't we set the env var before the call to `$(MAKE) dev-deploy-e2e`?

Comment (alexeykazakov) [reply make/dev.mk]: Doesn't matter. Both `MEMBER_REPO_PATH=... make dev-...` and `make dev-... MEMBER_REPO_PATH=... ` options work.

File: README.adoc [hunk 2]
```diff
@@ -115,3 +116,17 @@ ci-operator --config ../../openshift/release/ci-operator/config/codeready-toolch
...
+
+* `make dev-deploy-e2e-local` - deploys the same resources as `make test-e2e-local` but doesn't run tests.
+
+* `make dev-deploy-e2e` - deploys the same resources as `make test-e2e` but doesn't run tests.
+
+By default these targets deploy resources to `toolchain-host-operator` and `toolchain-member-operator` namespaces.
+
+NOTE: you can override the default namespace names via `make dev-deploy-e2e DEV_HOST_NS=my-host DEV_MEMBER_NS=my-member`
+
+NOTE: if running in Minishift `eval $(minishift docker-env)` `eval $(minishift oc-env)` is required and if running in CodeReady Containers: `eval $(crc oc-env)`
```
Comment (tinakurian) [inline README.adoc]:
```suggestion
NOTE: If running in Minishift `eval $(minishift docker-env)` and `eval $(minishift oc-env)` are required. If running in CodeReady Containers `eval $(crc oc-env)` is required.
```

File: README.adoc [hunk 3]
```diff
@@ -115,3 +116,17 @@ ci-operator --config ../../openshift/release/ci-operator/config/codeready-toolch
...
+
+All e2e resources (host operator, member operator, registration-service, CRDs, etc) can be deployed without running tests:
+
+* `make dev-deploy-e2e-local` - deploys the same resources as `make test-e2e-local` but doesn't run tests.
+
+* `make dev-deploy-e2e` - deploys the same resources as `make test-e2e` but doesn't run tests.
+
+By default these targets deploy resources to `toolchain-host-operator` and `toolchain-member-operator` namespaces.
+
+NOTE: you can override the default namespace names via `make dev-deploy-e2e DEV_HOST_NS=my-host DEV_MEMBER_NS=my-member`
```
Comment (tinakurian) [inline README.adoc]:
```suggestion
NOTE: You can override the default namespace names via `make dev-deploy-e2e DEV_HOST_NS=my-host DEV_MEMBER_NS=my-member`
```

**Pull request merged**